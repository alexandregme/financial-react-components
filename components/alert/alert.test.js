import React from 'react';
import { shallow, mount } from 'enzyme';
import Alert from 'components/alert';

describe('Alert', () => {
  let mountedAlert;
  let alertContainer;
  let closeButton;

  beforeEach(() => {
    mountedAlert = shallow(<Alert />);
    alertContainer = mountedAlert.find('div');
    closeButton = alertContainer.find('button[className="close"]');
  });

  it('renders without crashing', () => {
    shallow(<Alert />);
    expect(mountedAlert).toMatchSnapshot();
  });

  it('Contains a container div to hold the alert', () => {
    expect(alertContainer).toHaveLength(1);
    expect(alertContainer.prop('className')).toEqual('alert alert-danger');
    expect(alertContainer.text()).toEqual('×');
  });

  it('expect span to hold the message', () => {
    const spanMessage = alertContainer.find('span[className="message"]');
    expect(spanMessage).toHaveLength(1);
    expect(spanMessage.text()).toEqual('');
  });

  it('expect a close button inside alert container', () => {
    expect(closeButton).toHaveLength(1);
    expect(closeButton.text()).toEqual('×');
    expect(closeButton.props().type).toEqual('button');
    expect(closeButton.props().onClick()).toBe(undefined);
    const spanInsideButton = closeButton.find('span');
    expect(spanInsideButton).toHaveLength(1);
    expect(spanInsideButton.prop('aria-hidden')).toEqual('true');
  });

  it('call a function passed to it when clicked', () => {
    const mockCallBack = jest.fn();
    const mountedAlertWithCallback = shallow(<Alert onButtonClick={mockCallBack} />);
    mountedAlertWithCallback.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

describe('Alert custom props', () => {
  let mountedAlert;
  let alertContainer;

  const props = {
    alertMessage: 'custom label',
    className: 'custom',
    messageType: 'success',
    onButtonClick: () => 'onButtonClick',
  };

  beforeEach(() => {
    mountedAlert = mount(<Alert {...props} />);
    alertContainer = mountedAlert.find('div');
  });

  it('should receive a prop text and show it as label', () => {
    expect(alertContainer.text()).toEqual(`${props.alertMessage}×`);
  });

  it('expect to add a custom className to label', () => {
    expect(alertContainer.prop('className')).toEqual('alert custom alert-success');
  });
});
