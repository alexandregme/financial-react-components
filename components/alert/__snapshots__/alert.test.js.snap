// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders without crashing 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <span
    className="message"
  />
  <button
    className="close"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
</div>
`;
